src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,380 @@
+        if (executorService != null) {
+            executorService.submit(() -> processBatch(drafts));
+        } else {
+            processBatch(drafts);
+        }
+    }
+
+    /**
+     * When processed in parallel it awaits for all tasks from {@code executorService} to end.
+     */
+    private void awaitDraftsProcessingFinished() {
+        if (executorService != null) {
+            executorService.shutdown();
+            try {
+                executorService.awaitTermination(14, TimeUnit.DAYS);
+            } catch(InterruptedException ex) {
+                LOGGER.error("Parallel inventories sync process was interrupted.", ex);
+            }
+        }
+    }
+
+    /**
+     * Method process batch of {@link InventoryEntryDraft}. The processing means:
+     * <ol>
+     *     <li>Fetching existing {@link InventoryEntry} that correspond to passed {@code drafts}</li>
+     *     <li>Deciding if each draft should be used for creating new entry or updating existing one</li>
+     *     <li>Attempting to create new entries or update existing ones</li>
+     *     <li>Updates the {@code statistics} object backed for this instance</li>
+     * </ol>
+     * @param drafts batch of drafts that need to be synced
+     */
+    private void processBatch(final List<InventoryEntryDraft> drafts) {
+        final Map<SkuKeyTuple, InventoryEntry> existingInventories = fetchExistingInventories(drafts);
+        drafts.forEach(draft -> {
+            final SkuKeyTuple skuKeyOfDraft = SkuKeyTuple.of(draft);
+            if (existingInventories.containsKey(skuKeyOfDraft)) {
+                final InventoryEntry existingEntry = existingInventories.get(skuKeyOfDraft);
+                attemptUpdate(existingEntry, draft);
+            } else {
+                attemptCreate(draft);
+            }
+            statistics.incrementProcessed();
+        });
+    }
+
+    /**
+     * Returns mapping of {@link SkuKeyTuple} to {@link InventoryEntry} of instances existing in a database.
+     * Instances are fetched from API by skus, that corresponds to skus present in {@code drafts}.
+     *
+     * @param drafts {@link List} of drafts
+     * @return {@link SkuKeyTuple} to {@link InventoryEntry} of existing entries that correspond to passed
+     * {@code drafts} by sku comparision.
+     */
+    private Map<SkuKeyTuple, InventoryEntry> fetchExistingInventories(final List<InventoryEntryDraft> drafts) {
+        final Set<String> skus = extractSkus(drafts);
+        final List<InventoryEntry> existingEntries = inventoryService.fetchInventoryEntriesBySkus(skus);
+        return mapSkuAndKeyToInventoryEntry(existingEntries);
+    }
+
+    /**
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} where each draft contains its sku
+     * @return {@link Set} of distinct skus found in {@code inventories}
+     */
+    private Set<String> extractSkus(final List<InventoryEntryDraft> inventories) {
```

Comment:
@andrii-kovalenko-ct Yes, we expect. There could be few inventory entries per same sku but differ in supply channels.